### ISSUE TITLE
SIGSEGV on CPDF_ColorSpace::GetBufSize()

### ISSUE THREAD

**Issue by Vadixem** [paulovap/qtpdfium#22]
Hello, Paulo!
I am trying to test your library. I added it to my qt project, but when I try to test it work using your sample code in REDME.MD, it crashes. Consider this data

Stack trace:
1  CPDF_ColorSpace::GetBufSize            cpdf_colorspace.cpp       411 0x5555559be96a 
2  CPDF_ColorSpace::CreateBuf             cpdf_colorspace.cpp       418 0x5555559be99e 
3  CPDF_Color::SetColorSpace              cpdf_color.cpp            65  0x5555559bd60b 
4  CPDF_ColorState::ColorData::SetDefault cpdf_colorstate.cpp       152 0x5555559c33b7 
5  CPDF_ColorState::SetDefault            cpdf_colorstate.cpp       25  0x5555559c2ce2 
6  CPDF_ContentParser::Continue           cpdf_contentparser.cpp    170 0x5555559c64c3 
7  CPDF_PageObjectHolder::ContinueParse   cpdf_pageobjectholder.cpp 33  0x5555557b953b 
8  CPDF_Page::ParseContent                cpdf_page.cpp             99  0x5555557b7239 
9  FPDF_LoadPage                          fpdfview.cpp              636 0x5555557afbbc 
10 QPdfium::page                          qpdfium.cpp               109 0x5555557a9af2 
11 main                                   main.cpp                  46  0x5555556c5075 

```
int CPDF_ColorSpace::GetBufSize() const {
  if (m_Family == PDFCS_PATTERN) { // Crashes when debugger gets here
    return sizeof(PatternValue);
  }
  return m_nComponents * sizeof(FX_FLOAT);
}
```

main.cpp:

```
#include <QApplication>
#include <QtPdfium>

int main(int argc, char *argv[])
{
    QApplication a(argc, argv);

    QPdfium pdf;
     auto result = pdf.loadFile("path/to/some/file"); // Returns success
    auto pdfPage = pdf.page(0); // Crashes here 
    auto valid = pdfPage.isValid();
    return a.exec();
}
```

**Comment by Vadixem** [comment]
I might have just linked library the wrong way, so it cannot find .a file

**Comment by tschust** [comment]
@Vadixem I run into the same issue when rendering a pdf file in iOS. How did you fix it?

**Comment by Vadixem** [comment]
> 
> 
> @Vadixem I run into the same issue when rendering a pdf file in iOS. How did you fix it?

Hello @tschust. To be honest, I am not sure whether I fixed it, but I wrote 

> I might have just linked library the wrong way, so it cannot find .a file

so I would suggest you to make sure in .pro file that you have "LIBS += -L"/path/to/qtpdfium_lib_files" -llibname
Where -llibname is name of library file (be it .a or .dylib) without .a or .dylib part and without "lib" part.
For instance if we have usr/user/bin/libcoolstuff.a in  .qmake file it will turn into
"LIBS += -L"usr/user/bin" -lcoolstuff

Hope it helps!
